stl/inc/type_traits

```diff
@@ -1695,6 +1699,7 @@ struct _Invoke_traits<void_t<decltype(_STD invoke(_STD declval<_Types>()...))>,
         conjunction_v<_Is_nothrow_invocable, disjunction<is_void<_Rx>, _Is_nothrow_convertible<type, _Rx>>>>;
 };
 
+#if !_HAS_DEPRECATED_TYPETRAITS

```

Comment:
You're testing `#if !_HAS_DEPRECATED_TYPETRAITS` but it seems to me that this is reversed. After looking at the definition (not just the naming), I believe you intended to guard this machinery with `#if _HAS_DEPRECATED_TYPETRAITS`. Am I completely confused?

Reply:
Nope that was me who missed the double negation. fixed